Implement useQueueStatus

marcoskwkm opened the pull request:
#### What problem is this solving?

Both `OrderItems` and `OrderShipping` want to perform actions conditioned on the status of the queue. In order to do that, they implemented a `ref` object controlled by the events emitted by the queue. This PR moves this logic to `OrderQueue` to avoid this copy-pasta.

In short, this adds a new helper hook `useQueueStatus` that receives the `listen` function returned by `useOrderQueue()` and returns a `ref` object containing the status of the queue (either `Pending` or `Fulfilled`).

#### How should this be manually tested?

`yarn test`
[This workspace](http://orderstatus--vtexgame1.myvtex.com/cart) can also be used for testing. I've linked to it a [modified `OrderItems`](https://github.com/vtex-apps/order-items/tree/chore/queue-status) that uses this new hook.

#### Checklist/Reminders

- [x] Updated `README.md`.
- [x] Updated `CHANGELOG.md`.
- [ ] Linked this PR to a Clubhouse story (if applicable).
- [x] Updated/created tests (important for bug fixes).
- [ ] Deleted the workspace after merging this PR (if applicable).

#### Screenshots or example usage

#### Type of changes

<!--- Add a ✔️ where applicable -->
✔️ | Type of Change
---|---
_ | Bug fix <!-- a non-breaking change which fixes an issue -->
✔️ | New feature <!-- a non-breaking change which adds functionality -->
_ | Breaking change <!-- fix or feature that would cause existing functionality to change -->
_ | Technical improvements <!-- chores, refactors and overall reduction of technical debt -->

#### Notes

<!-- Put any relevant information that doesn't fit in the other sections here. -->

darlenedms commented on react/modules/TaskQueue.ts:
```diff
@@ -9,6 +9,11 @@ export const TASK_CANCELLED_MSG =
+export const STATUS: Record<string, QueueEvent> = {
+  PENDING: 'Pending',
+  FULFILLED: 'Fulfilled',
+}
```
⛏ Consider moving it to a constants file.

marcoskwkm replied:
This was exported only to be used in the tests, so I don't see much value in moving this to a separate file for now.

marcoskwkm replied:
Ok, I lied, these values are used both by `OrderQueue` and the `TaskQueue` implementation itself. I transformed this into an enum, moved to a constants file and exported it with `OrderQueue`, so it can be used by the consuming apps.

marcoskwkm commented:
I moved the `STATUS` object to a constants file, exported it with `OrderQueue` and updated the docs. Since those were not trivial changes I'm requesting review again (sorry!). The list of changes before and after the new changes are [here](https://github.com/vtex-apps/order-manager/compare/e2751fabfb796052fb6f0b6d852190da500a4da5..1999d4306bf11f801269031fc6a5c4821c89ed30).
